File: mobile-app/lib/models/exercises.model.dart
class Exercise {
  final String id;
  final String title;
  final String derscription;
  final String videoId;
  final String category;
  final String duration;
  final String thumbnailUrl;
  final DateTime createdAt;
  final DateTime updatedAt;

  Exercise({
    required this.id,
    required this.title,
    required this.derscription,
    required this.videoId,
    required this.category,
    required this.duration,
    required this.thumbnailUrl,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json["_id"],
      title: json["title"],
      derscription: json["derscription"] ?? "",
      videoId: json["videoId"],
      category: json["category"],
      duration: json["duration"],
      thumbnailUrl: json["thumbnailUrl"],
      createdAt: DateTime.parse(json["createdAt"]),
      updatedAt: DateTime.parse(json["updatedAt"]),
    );
  }
}

File: backend/models/Exercises.model.js
const { required } = require("joi");
const mongoose = require("mongoose");

const exerciseSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      required: true,
      trim: true,
    },
    videoId: {
      type: String,
      required: true,
      trim: true,
      match: [
        /^[a-zA-Z0-9_-]{11}$/,
        "The video ID must be an 11 character EX of :YouTube video ID (dQw4w9WgXcQ)",
      ],
    },
    category: {
      type: String,
      enum: [
        "strength",
        "stretch",
        "relaxation",
        "balance",
        "mobility",
        "other",
      ],
    },
    duration: {
      type: String,
      required: true,
    },
    thumbnailUrl: {
      type: String,
      required: true,
      trim: true,
    },
  },
  {
    timestamps: true,
  }
);

const Exercise = mongoose.model("Exercise", exerciseSchema);

module.exports = Exercise;
